akka-persistence/src/main/scala/akka/persistence/journal/AsyncWriteJournal.scala

```diff
@@ -69,7 +69,8 @@ trait AsyncWriteJournal extends Actor with WriteJournalBase with AsyncRecovery {
         val atomicWriteCount = messages.count(_.isInstanceOf[AtomicWrite])
         val prepared = Try(preparePersistentBatch(messages))
         val writeResult = (prepared match {
-          case Success(prep) =>
+          case Success(prep) if prep.isEmpty => Future.successful(Nil)
```

Comment:
It's not obvious why we need it. For instance, what 'prepare' means here? And why we should not pass an empty `prep` to `asyncWriteMessages`? 

Can we add a block of comment explaining what is happening and why we need it?

Reply:
see update and let me know if that made things clearer. Thanks!

Reply:
Yes, it is. 
Especially that part: `most plugins fail calling head on empty Seq`

I don't think that was ever said in the discussion here. I was assuming it, but now it's confirmed and written down. Thanks!